src/ol_infrastructure/applications/mitxonline/__main__.py

```diff
@@ -0,0 +1,164 @@
+"""
+import json
+
+import pulumi_vault as vault
+from pulumi import Config, StackReference, export
+from pulumi_aws import ec2, iam, s3
+
+from ol_infrastructure.components.aws.database import OLAmazonDB, OLPostgresDBConfig
+from ol_infrastructure.components.services.vault import (
+    OLVaultDatabaseBackend,
+    OLVaultPostgresDatabaseConfig,
+)
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.lib.stack_defaults import defaults
+
+mitxonline_config = Config("mitxonline")
+stack_info = parse_stack()
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+apps_vpc = network_stack.require_output("applications_vpc")
+operations_vpc = network_stack.require_output("operations_vpc")
+aws_config = AWSBase(
+    tags={
+        "OU": "mitxonline",
+        "Environment": f"applications_{stack_info.env_suffix}",
+        "Application": "mitxonline",
+    }
+)
+
+# Create S3 bucket
+
+# Bucket used to store files from MITx Online app.
+mitxonline_bucket_name = f"ol-mitxonline-app-{stack_info.env_suffix}"
+mitxonline_bucket = s3.Bucket(
+    f"mitxonline-{stack_info.env_suffix}",
+    bucket=mitxonline_bucket_name,
+    versioning=s3.BucketVersioningArgs(
+        enabled=True,
+    ),
+    tags=aws_config.tags,
+)
+
+
+mitxonline_iam_policy = iam.Policy(
+    f"mitxonline-{stack_info.env_suffix}-policy",
+    description="AWS access controls for the MITx Online application in the "
+    f"{stack_info.name} environment",
+    path=f"/ol-applications/mitxonline/{stack_info.env_suffix}/",
+    name_prefix="aws-permissions-",
+    policy=lint_iam_policy(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "s3:ListAllMyBuckets",
+                    "Resource": "*",
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:ListBucket*",
+                        "s3:PutObject",
+                        "s3:PutObjectAcl",
+                        "s3:GetObject*",
+                        "s3:DeleteObject*",
+                    ],
+                    "Resource": [
+                        f"arn:aws:s3:::{mitxonline_bucket_name}",
+                        f"arn:aws:s3:::{mitxonline_bucket_name}/*",
+                    ],
+                },
+            ],
+        },
+        stringify=True,
+        parliament_config={
+            "PERMISSIONS_MANAGEMENT_ACTIONS": {
+                "ignore_locations": [{"actions": ["s3:putobjectacl"]}]
+            }
+        },
+    ),
+)
+
+mitxonline_vault_backend_role = vault.aws.SecretBackendRole(
+    f"mitxonline-app-{stack_info.env_suffix}",
+    name=f"mitxonline-{stack_info.env_suffix}",
```

Comment:
We don't want the role name to have any indication of environment. That prevents the application from being naive as to where it's running. Each Vault cluster will have the role defined separately.